### PR TITLE
 Update Compatibility Rules for ENUM Extensions ST

### PR DESCRIPTION
**Pull request by burcuka**

### CODE REVIEW

File: src/zcl_aff_writer_xslt.clas.abap [hunk 1]
```diff
@@ -765,4 +787,30 @@ CLASS zcl_aff_writer_xslt IMPLEMENTATION.
...
+      ELSE.
+        xml_value = <enum_value>-overwritten_json_value.
+      ENDIF.
+      IF index < lines( enum_values ).
+        write_open_tag( |<tt:cond-var check="VARIABLE='{ xml_value }'">| ).
+        write_tag( |<tt:assign { get_to_ref( element_name ) } val="{ abap_value }"/>| ).
+        write_closing_tag( `</tt:cond-var>` ).
+      ELSE.
+        write_open_tag( |<tt:cond-var check="VARIABLE={ xml_value }| ).
+        write_tag( `"/>` ).
```
Comment (wurzka) [inline src/zcl_aff_writer_xslt.clas.abap:809]: I think we can delete this whole index handling here. So just keep the code in the if-part and remove the variable index

File: src/zcl_aff_writer_xslt.clas.abap [hunk 2]
```diff
@@ -765,4 +787,30 @@ CLASS zcl_aff_writer_xslt IMPLEMENTATION.
...
+      ENDIF.
+      index += 1.
+    ENDLOOP.
+    write_closing_tag( ` </tt:deserialize>` ).
```
Comment (wurzka) [inline src/zcl_aff_writer_xslt.clas.abap:813]: The space in front of < should be deleted here

File: src/zcl_aff_writer_xslt.clas.abap [hunk 3]
```diff
@@ -304,17 +314,23 @@ CLASS zcl_aff_writer_xslt IMPLEMENTATION.
...
       write_iso_language_callback( element_name = element_name ).
     ELSEIF enum_values IS INITIAL.
       write_tag( |<tt:value{ get_ref( element_name ) }{ get_option( json_type = type element_description = element_description ) }/>| ).
     ELSE.
-      write_enum_value_mappings( element_description = element_description json_type = type element_name = element_name enum_values = enum_values ).
+      write_enum_map_ext_compatible(
+        element_description = element_description
+        element_name        = element_name
+        enum_values         = enum_values ).
+      write_open_tag( |<tt:serialize>| ).
```
Comment (wurzka) [inline src/zcl_aff_writer_xslt.clas.abap:328]: The deserialize-tag is written in the write_enum_map_ext_compatible, the serializa-tag outside. Should we keep this in sync? Maybe both outside? What do you think?

**Review by wurzka — CHANGES_REQUESTED** [review]
Thanks :) See my comments

File: src/zcl_aff_writer_xslt.clas.abap (hunk 3 above)
Comment (burcuka) [reply src/zcl_aff_writer_xslt.clas.abap:328]: yeah, i think outside is better

File: src/zcl_aff_writer_xslt.clas.abap [hunk 4]
```diff
@@ -765,4 +789,22 @@ CLASS zcl_aff_writer_xslt IMPLEMENTATION.
 
   ENDMETHOD.
 
+
+  METHOD write_enum_map_ext_compatible.
+    write_tag( line = |<tt:read type="C" var="VARIABLE"/>| ).
+    DATA index TYPE i.
```
Comment (wurzka) [inline src/zcl_aff_writer_xslt.clas.abap:795]: We don't need it anymore, no?

**Review by wurzka — CHANGES_REQUESTED** [review]
Just one minor, else looks good

File: src/zcl_aff_writer_xslt.clas.abap (hunk 4 above)
Comment (burcuka) [reply src/zcl_aff_writer_xslt.clas.abap]: sure ;)

**Review by wurzka — APPROVED** [review]
Thanks, LGTM :)

**Pull request merged**